### ISSUE TITLE
zero firing rate units in `z_score_signal`

### ISSUE THREAD

**Issue by AtMostafa** [mattperich/PyalData#107]
In `z_score_signal` ([here](https://github.com/mattperich/PyalData/blob/f0562256bdf5613b5977254754135f52dbf9657b/pyaldata/tools.py#L354)), there should be a check for units without any spikes, otherwise dividing by `np.std` causes NaN values and problems downstream.

**Comment by bagibence** [comment]
Wouldn't calling `remove_low_firing_neurons` before z-scoring eliminate this?

I'm a bit uncomfortable with #108. I think I'd prefer throwing errors/warnings or dealing with NaNs rather than silently modifying the data.

**Comment by AtMostafa** [comment]
you mean adding something like `remove_low_firing_neurons(df,signal, 0.01)` in the `z_score_signal` function?  
Implicitly expecting the user to have run `remove_low_firing_neurons` beforehand is not ideal and finding the source of the error takes a long time if later analyses can't handle NaNs.  
BTW, is it really modifying the data? I guess if later analyses depend on having _std==1_, you're right!

**Comment by bagibence** [comment]
No, I think the z-scoring function should just do the one thing it's supposed to.

I think numpy warns you when you're dividing by zero, right? If not, we could warn the user that they have some quiet neurons.

I start every notebook with `remove_low_firing_neurons` (but then never z-score).
All-zeros in some columns mess up many analyses, too, which is what you get if you leave non-spiking neurons in.

By modifying the data I mostly meant we're not applying the same transformation to every unit.

**Comment by AtMostafa** [comment]
I agree with the warning, let's just leave the rest as is.

**Comment by bagibence** [comment]
I just checked and numpy gives this warning by default:
> RuntimeWarning: invalid value encountered in true_divide
  trial_data[signal] = [(s - col_mean) / col_std for s in trial_data[signal]]

Is this okay or do we want to add another warning ourselves?

**Comment by AtMostafa** [comment]
I wonder why I didn't get any warnings then... anyway, as long as there is some indication, it should be ok.
